modules/post/windows/manage/install_python.rb

```diff
@@ -0,0 +1,84 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'          => 'Install Python for Windows',
+                      'Description'   => '
+                        This module places an embeddable Python3 interpreter onto the target file system,
+                        granting pentesters access to a lightweight Python interpreter.
+                        This module does not require administrative privileges or user interaction with
+                        installation prompts.
+                      ',
+                      'License'       => MSF_LICENSE,
+                      'Author'        => ['Michael Long <bluesentinel[at]protonmail.com>'],
+                      'Arch' => [ARCH_X86, ARCH_X64],
+                      'Platform'      => [ 'win' ],
+                      'SessionTypes'  => [ 'meterpreter', 'shell' ],
+                      'References'	=> [
+                        ['URL', 'https://docs.python.org/3/using/windows.html#windows-embeddable'],
+                        ['URL', 'https://attack.mitre.org/techniques/T1064/']
+                      ]))
+    register_options(
+      [
+        OptString.new('PYTHON_URL', [true, 'Python embeddable zip URL', 'https://www.python.org/ftp/python/3.8.2/python-3.8.2-embed-win32.zip']),
+        OptString.new('FILE_PATH', [true, 'File path to store the python zip file; current directory by default', '.\\python-3.8.2-embed-win32.zip']),
+        OptBool.new('CLEANUP', [false, 'Remove module artifacts; set to true when ready to cleanup', false])
+      ]
+    )
+  end
+
+  def run
+    python_folder_path = File.basename(datastore['FILE_PATH'].to_s, File.extname(datastore['FILE_PATH'].to_s))
+    python_exe_path = python_folder_path + "\\python.exe"
+
+    # check if PowerShell is available
+    psh_path = "\\WindowsPowerShell\\v1.0\\powershell.exe"
+    if !file? "%WINDIR%\\System32#{psh_path}"
+      fail_with(Failure::NotVulnerable, "No powershell available.")
+    end
```

Comment:
As you're using the Powershell mixin, you could probably use `have_powershell?` here.

```ruby
 28         #
 29         # Returns true if powershell is installed
 30         #
 31         def have_powershell?
 32           cmd_exec('cmd.exe', '/c "echo. | powershell get-host"') =~ /Name.*Version.*InstanceId/m
 33         end
```

```suggestion
    # check if PowerShell is available
    unless have_powershell?
      fail_with(Failure::NotVulnerable, "No PowerShell available.")
    end
```